src/ROOT-Sim.h

```diff
@@ -71,7 +71,7 @@ typedef bool (*CanEnd_t)(lp_id_t me, const void *snapshot);
 /// These event types are automatically scheduled to the model according to the following logic:
 /// - `LP_INIT`: Represents the initialization event for a logical process.
 /// - `LP_FINI`: Represents the finalization event for a logical process.
-enum rootsim_event { LP_INIT = 65534, LP_FINI };
+enum rootsim_event { LP_RETRACTABLE = 0, LP_INIT = 65534, LP_FINI };
```

Comment:
We should decide whether to deliver retractables before or after other events.
Personally, I would give them higher priority (setting LP_RETRACTABLE = 65533) and deliver them before user events.
```suggestion
enum rootsim_event { LP_RETRACTABLE = 65533, LP_INIT = 65534, LP_FINI };
```